### PR TITLE
Test: Data type precedence, conversion NVARCHAR(n)

### PR DESCRIPTION
**Pull request by v-kigos**

### CODE REVIEW

**Comment by v-dareck** [comment]
For srv_029_data_conversion_varchar_datetime.phpt can the commented out test lines be removed.

**Comment by v-kigos** [comment]
Done

File: test/sqlsrv/srv_029_data_conversion_varchar_datetime.phpt [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+// Insert data. The data type with the lower precedence is 
+// converted to the data type with the higher precedence
+$sql = "INSERT INTO $tableName VALUES (''),(-378.4),(43000.4),(GETDATE())";
+$stmt = sqlsrv_query($conn, $sql);
+
+// Read data from the table
+$sql = "SELECT * FROM $tableName";
+$stmt = sqlsrv_query($conn, $sql);
+for($i=0; $i<3; $i++)
```
Comment (yitam) [inline test/sqlsrv/srv_029_data_conversion_varchar_datetime.phpt]: Is there any reason why you have inserted four values but only fetching three back?

File: test/sqlsrv/srv_029_data_conversion_varchar_datetime.phpt [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+--FILE--
+<?php
+
+require_once("autonomous_setup.php");
+
+// Connect
+$conn = sqlsrv_connect($serverName, $connectionInfo) ?: die();
+
+// Create database
+sqlsrv_query($conn,"CREATE DATABASE ". $dbName) ?: die();
```
Comment (yitam) [inline test/sqlsrv/srv_029_data_conversion_varchar_datetime.phpt]: It seems a little over the top to create a database for one simple test... is it possible that you create a table with some sort of random number to guarantee its uniqueness?

File: test/sqlsrv/srv_029_data_conversion_varchar_datetime.phpt (hunk 1 above)
Comment (v-kigos) [inline test/sqlsrv/srv_029_data_conversion_varchar_datetime.phpt]: Sure. To get it less straightforward, and possibly discover new issues.

File: test/sqlsrv/srv_053_mars_disabled_error_checks.phpt [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            [2] => The connection cannot process this operation because there is a statement with pending results.  To make the connection available for other queries, either fetch all results or cancel or free the statement.  For more information, see the product documentation about the MultipleActiveResultSets connection option.
+            [message] => The connection cannot process this operation because there is a statement with pending results.  To make the connection available for other queries, either fetch all results or cancel or free the statement.  For more information, see the product documentation about the MultipleActiveResultSets connection option.
+        )
+
+    [1] => Array
+        (
+            [0] => HY000
+            [SQLSTATE] => HY000
+            [1] => 0
+            [code] => 0
```
Comment (yitam) [inline test/sqlsrv/srv_053_mars_disabled_error_checks.phpt]: Are the error messages the same in both Windows and Linux?

Comment (yitam) [inline test/sqlsrv/srv_053_mars_disabled_error_checks.phpt]: Also, I assume the errors come from the second call to sqlsrv_query() not the first sqlsrv_fetch()?

Comment (v-kigos) [inline test/sqlsrv/srv_053_mars_disabled_error_checks.phpt]: yes

**Pull request merged**